src/duk_debugger.c

```diff
@@ -64,6 +79,8 @@ DUK_LOCAL void duk__debug_do_detach1(duk_heap *heap) {
 	 * XXX: clear breakpoint on either attach or detach?
 	 */
 	heap->dbg_breakpoints_active[0] = (duk_breakpoint *) NULL;
+
+	heap->dbg_detaching = 0;
```

Comment:
One perhaps more robust alternative for this flag would be "detach in progress or completed" (not sure how to name it shortly). The difference would be that we'd leave the flag to 1 until the next attach, so that any code path ending up trying to detach again (perhaps because of a bug) wouldn't unnecessarily go through the motions.